### DISCUSSION TITLE
Extend execution period for submissions

### DISCUSSION THREAD

**Discussion by begovegar** [PaddlePaddle/PaddleSpatial#185] [BaiduKDDCup2022] [answered]
Hi, we've tried to make different submissions that eventually failed due to this error:
![image](https://user-images.githubusercontent.com/104911705/174277192-49989f8a-7be0-4b29-8f7e-f06b6ff985b4.png)
We've seen in the code that if the training time exceeds 10h the execution stops. Is it possible to extend this period so that our execution finishes? It works in our local pc so I guess it should in yours. 
Thanks a lot!

**Comment by ramber1836** [comment]
Please provide your team name.

**Comment by begovegar** [comment]
Sure! We are Innova Team.

**Reply by ramber1836** [reply]
The error log is as follows: 

![image](https://user-images.githubusercontent.com/19561583/174320125-03c17996-f53a-4d59-ad18-a81ca79e1288.png)

**Comment by begovegar** [comment, marked as answer]
Sorry we don't understand what does the red squared text mean. (Get text x in 761.02 secs) What's the issue?

**Reply by ramber1836** [reply]
It might be caused by the global smoothing you adopted.

**Reply by sergiosmg96** [reply]
But there is a mismatch between our time estimations in our local environment and the real execution time. We estimate a total execution time lower than 10 hours. However, the system does not allow us to complete our submissions.

**Reply by ramber1836** [reply]
The online test predicts one instance (read the test_x and forecast test_y) at a time and repeats this procedure for 195 instances. Does your local evaluation follow the same paradigm? If not, the estimated run time in your local machine is not consistent with the online test. 
Besides, the efficiency of your code should be optimized. 
Sorry, we cannot extend the TIMEOUT under the current settings.

**Reply by begovegar** [reply]
Yes, our evaluation follow the same paradigm and each instance takes betwwen 70 and 175 secs. And those above 150 secs didn't finish due to the timeout error. We're trying to optimize our code but anyway, those executions lasting 175 secs. per k should finish, dont you think so?
Thanks.

**Reply by ramber1836** [reply]
Yes, if the prediction time of your code is no more than 3 mins then it should be OK your code will finish on time. 
BTW, the efficiency of the submitted code is one important factor when the final evaluation begins.

**Reply by begovegar** [reply]
Sorry but if our prediction time for each instance is 175 secs
Why in your system each instance takes 760-780secs? Is this what the error log says?

**Reply by ramber1836** [reply]
Many factors may contribute to that outcome, such as different hardware, online scheduling, and different package environments. 
You can make a detailed time consumption assessment for the global smoothing of your code by running, like, 10 predictions, and raising an fake exception for debugging your code. Obviously, this will increase our workload as you will ask us to post the detailed logs of your submitted code. 
Hope you can figure out how to optimize your code on your local machine.
